qclib/reversed_gates.qc

```diff
@@ -0,0 +1,13 @@
+
+
+; rev_CNOT implements a CNOT gate where the
+; the control qbit `a` have a higher qbit number
+; Question: This rather feels like a limitation
+; in the driver, maybe we should build a generic
+; reversibility there instead?
+macro rev_CNOT a b
```

Comment:
Also, I think rev_CNOT is not a perfect name since it somehow sounds like the inverse of CNOT. I started a little bit on a stdlib.qc and there I call it udCNOT for up-side-down.